.github/workflows/ci.yml

```diff
@@ -15,8 +15,6 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
-        with:
-          ref: ${{ github.head_ref }}
```

Comment:
I'm a total newbie to github actions, could you please explain why this was making the `git fetch` fail? How does `actions/checkout` pick the branch to checkout now?

Reply:
@hernando I'm not super familiar with GH actions either, but the [docs here](https://github.com/actions/checkout/blob/85e6279cec87321a52edac9c87bce653a07cf6c2/README.md?plain=1#L27-L30) explain the default behavior better than I can.

It seems like `github.head_ref` was invalid when the PR was coming from the branch of a fork of the repo.
```    
    # The branch, tag or SHA to checkout. When checking out the repository that
    # triggered a workflow, this defaults to the reference or SHA for that event.
    # Otherwise, uses the default branch.
```